src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,325 @@
+};
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  NormalizeTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename InputT>
+NormalizeTransformer<InputT>::NormalizeTransformer(std::vector<std::double_t> norms) :
+    _norms(std::move(norms)),
+    _row(0) {
+        for (auto it = norms.begin(); it != norms.end(); ++it) {
+            if (*it < 0) {
+                throw std::invalid_argument("Norms shouldn't be less than 0!");
+            }
+        }
+}
+
+template <typename InputT>
+NormalizeTransformer<InputT>::NormalizeTransformer(Archive &ar) :
+    NormalizeTransformer(
+        [&ar](void) {
+            std::vector<std::double_t> norms(Traits<std::vector<std::double_t>>::deserialize(ar));
+
+            return NormalizeTransformer<InputT>(std::move(norms));
+        }()
+    ) {
+}
+
+template <typename InputT>
+void NormalizeTransformer<InputT>::save(Archive &ar) const /*override*/ {
+    Traits<std::vector<std::double_t>>::serialize(ar, _norms);
+}
+
+template <typename InputT>
+bool NormalizeTransformer<InputT>::operator==(NormalizeTransformer const &other) const {
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#endif
+
+    return this->_norms == other._norms;
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#endif
+
+}
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename InputT>
+void NormalizeTransformer<InputT>::execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) /*override*/ {
+    execute_impl(input, callback, std::integral_constant<bool, Microsoft::Featurizer::Traits<ValueType>::IsNullableType>());
+}
+
+template <typename InputT>
+void NormalizeTransformer<InputT>::execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback, std::true_type) {
+    // ----------------------------------------------------------------------
+    using InputTraits                       = Traits<ValueType>;
+    using TransformedTraits                 = Traits<long double>;
+    // ----------------------------------------------------------------------
+    std::vector<long double> res;

```

Comment:
>res [](start = 29, length = 3)

Since you know what the size of this vector is going to be, prereserve the space via `res.reserve(input.size());`